packages/css/src/plugins/discard-duplicates.tsx

```diff
@@ -1,15 +1,14 @@
-import type { Declaration } from 'postcss';
-import { plugin } from 'postcss';
+import type { Plugin, Declaration } from 'postcss';
 
 /**
  * Discards top level duplicate declarations.
  */
-export const discardDuplicates = plugin<{ callback: (sheet: string) => void }>(
-  'discard-duplicates',
-  () => {
-    const decls: Record<string, Declaration[]> = {};
+export const discardDuplicates = (): Plugin => {
+  return {
+    postcssPlugin: 'discard-duplicates',
+    Once(root) {
+      const decls: Record<string, Declaration[]> = {};
```

Comment:
super nit: This can be a returning arrow function now

Reply:
I might introduce a new eslint rule for this and sort it out in a different PR